### DISCUSSION TITLE
Dexcom Share - Can I specify the seconds before polling retry?

### DISCUSSION THREAD

**Discussion by utechnet** [Navid200/xDrip#75] [Dexcom]
Hi, I am a user of Xdrip+ with my wife connected to our T1D son who wears a Dexcom G6 and is receiving readings via Dexcom app on his Pixel 3a, then shared to Dex Share. His phone reads Xdrip+ as a Dex Share follower, same as mine and my wife's. His endocrinologist wants to be shared the data with Dexcom Clarity so we need to do it this way.  We fairly consistently have an issue where there is a delay in his phone getting the reading uploaded to the Dexcom Share server which therefore creates an issue with ours being able to pull that reading down within the constraints of the app. I realize by watching the Dex Share Follow status screen that it polls for a new reading from the Dex Share server every 5 minutes and if it doesn't receive a reading it'll retry and poll again after 20 seconds. That 20 seconds seems like it's just barely not long enough about 30-40% of the time. I have learned that I can go to the Classic Status Page and press the "Restart Collector" button to have it poll right away if I really need to. Is there a way I can customize the retry delay timer or perhaps request a second retry if the first doesn't succeed? Thank you.

**Comment by Navid200** [comment]
@tolot27 
Is this related to https://github.com/NightscoutFoundation/xDrip/issues/929?
Or, do we need a new issue for this?

**Reply by utechnet** [reply]
I wasn't aware that issue existed.  After reading it over, it is remarkably similar to the issue that I experience.  The difference is that my source is Dexcom Share instead of Nightscout.  Would be awesome if I could specify the retry delay timer when BG isn't successfully retrieved after the initial polling.  Or perhaps a checkbox to enable a double retry (first after 20 seconds and second after 40 seconds).

**Reply by Navid200** [reply]
@tolot27 knows a lot more about this than me.  That's why I have asked for his advice.

**Reply by tolot27** [reply]
I'll check whether Dexcom Share Follow is using the same anticipation algorithm to calculate the new retry time. Anyway, it looks like something which can be improved.

**Comment by Navid200** [comment]
@utechnet If you wish, I guess you can open a new feature request.

If it turns out, after @tolot27 get's a chance to check, that the existing issue will address this as well, we will close your new feature request as it will effectively be redundant.  
On the other hand, if it turns out that the existing issue has nothing to do with Dexcom share, it will be good for others who may face the issues you are facing to be able to consolidate their requests in a dedicated issue.

So, as much as I hate opening new issues, if you wish, please go ahead.
